classes/board.js

```diff
@@ -7,6 +7,36 @@ class Board {
         this.width = data.width;
         this.food = data.food.map(c => new Coordinate(c));
         this.snakes = data.snakes.map(s => new Snake(s));
+        // Initialize 2D array
+        this.data = [];
+        for (let i = 0; i < this.width; i++) {
+            this.data.push([]);
+            for (let j = 0; j < this.height; j++) {
+                this.data[i].push(true);
+            }
+        }
+        // Fill grid with snake data
+        this.snakes.forEach( snake => {
+            snake.body.forEach( c => {
+                this.data[c.x][c.y] = false;
+            });
+            // Embed the snake's head on the board
+            this.data[snake.head.x][snake.head.y] = snake;
+        });
+    }
+    /*
+     * Returns:
+     *  True if the space is empty
+     *  False if the space is occupied or off the boad, but not a snake's head
+     *  A reference to the snake if the space is the snake's head
+     */
+    getValueAt(x, y) {
+        if (x >= 0 || x < this.width) {
```

Comment:
`x >= 0 || x < this.width`
shouldn't this be &&? want it to be greater than 0 AND less than the width, otherwise, it will always pass

The nested ifs is also bad practice, if you want to keep your conditions seperate for readability, I would do something like this:

```
let xWithinBounds = x >= 0 && x < this.width;
let yWithinBounds = y >= 0 && y < this.height;

if (xWithinBounds && yWithinBounds) {
    return this.data[x][y];
} else {
    return false;
}
```

and if you rly like being fancy, can simplify the return and get rid of the if altogether
`return xWithinBounds && yWithinBounds ? this.data[x][y] : false;`

Reply:
Fixed, thanks for the feedback!